src/pipeline_migration/actions/add_task.py

```diff
@@ -0,0 +1,406 @@
+
+    pipeline-migration-tool add-task push-dockerfile \\
+        /path/to/repo1/.tekton/pr.yaml /path/to/repo2/.tekton/push.yaml
+
+* Add task with parameter and execution order:
+
+    pipeline-migration-tool add-task push-dockerfile \\
+        --param param1,value1 --param param2,value2 \\
+        --run-after build-image-index
+
+* Add task with specific bundle reference:
+
+    pipeline-migration-tool add-task --bundle-ref <bundle-reference> push-dockerfile
+"""
+
+
+class InconsistentBundleBuild(Exception):
+    """Registry does not have expected bundle image"""
+
+
+class KonfluxTaskNotExist(Exception):
+    """Konflux task is not found from build-definitions"""
+
+
+class KonfluxTaskFileNotExist(Exception):
+    """Konflux task file does not exist in a version-specific task directory"""
+
+
+def konflux_task_bundle_reference(value: str) -> str:
+    """Argument type for checking input bundle reference
+
+    :raises argparse.ArgumentTypeError: if input bundle reference is invalid.
+    """
+    build_def = KonfluxBuildDefinitions()
+    try:
+        build_def.validate_bundle_ref(value)
+    except ValueError as e:
+        raise ArgumentTypeError(str(e))
+    return value
+
+
+def task_param(value: str) -> tuple[str, str]:
+    parts = value.split(",", 1)
+    if len(parts) == 1:
+        raise ArgumentTypeError("Missing parameter name or value.")
+    return parts[0], parts[1]
+
+
+def register_cli(subparser) -> None:
+    add_task_parser = subparser.add_parser(
+        "add-task",
+        help="Add a Konflux task to build pipelines.",
+        description=SUBCMD_DESCRIPTION,
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+    add_task_parser.add_argument(
+        "task",
+        help="Konflux task name to add. This is the actual task name defined in "
+        "konflux-ci/build-definitions. If a trusted artifact task is being added, "
+        "ensure the task name has suffix -oci-ta, e.g. buildah-oci-ta.",
+    )
+    add_task_parser.add_argument(
+        "file_or_dir",
+        nargs="*",
+        help="Specify pipeline files to where the task is added. "
+        "A pipeline can be included in a PipelineRun definition or a single Pipeline itself. "
+        "Directory path can also be specified alternatively. %(prog)s searches pipelines "
+        "from all YAML files from the first level of the directory.",
+    )
+    add_task_parser.add_argument(
+        "-n",
+        "--pipeline-task-name",
+        metavar="NAME",
+        help="Task name used in pipeline to configure the added task, in case it should be "
+        "different from the task name inside the bundle. For example, build task has name "
+        "build-container in the pipeline, but the actual task name is buildah inside bundle. "
+        "This is also useful for adding trusted artifact tasks, e.g. name the task as "
+        "sast-coverity-check but the actual task name is sast-coverity-check-oci-ta.",
```

Comment:
Would it make sense to remove the `-oci-ta` suffix automatically so that users don't have to remember to use `--pipeline-task-name`?

Reply:
I'd like to make it explicit. Users have to learn the command line interface in advance even if users don't have to remember to use `--pipeline-task-name`. On the other hand, for a trusted artifact task, the pipeline task name without `-oci-ta` is a convention but not forced.

Reply:
Seems reasonable. What about a `--strip-oci-ta-suffix` flag?

```
pipeline-migration-tool add-task sast-snyk-check-oci-ta --strip-oci-ta-suffix
pipeline-migration-tool add-task sast-cov-check-oci-ta --strip-oci-ta-suffix
```

Would be more convenient than

```
pipeline-migration-tool add-task sast-snyk-check-oci-ta --pipeline-task-name sast-snyk-check
pipeline-migration-tool add-task sast-cov-check-oci-ta --pipeline-task-name sast-cov-check
```

While still being explicit

Reply:
`--strip-oci-ta-suffix` is still bound to the specific `-oci-ta` handling. `--pipeline-task-name` is generic and its meaning is direct. Let's see the build container task. The pipeline task name is `build-container` and the actual task name is `buildah*`. Such tasks can't be added without generic option `--pipeline-task-name`. IMO, the CLI option should be generic enough rather than being specific to `-oci-ta`.

Reply:
Definitely +1 to keep `--pipeline-task-name`, but can we have both?

Reply:
I don't get your point. If users don't want  a pipeline task name with suffix `-oci-ta`, they just set the name to `--pipeline-task-name`, why tool has to provide a dedicated option to remove that suffix by code?

Reply:
Because it's a common Konflux convention. In our pipeline definitions, tasks with the `-oci-ta` suffix _always_ use a pipeline task name without the suffix.

If we don't make it extra easy for users to follow that convention, we'll get inconsistency and that will bring problems. For example, we'll want to write a migration script that passes `$(tasks.some-new-task.results.foo)` to another task. But now many users don't have `some-new-task` in their pipeline because they forgot to use `--pipeline-task-name` and they have `some-new-task-oci-ta` instead, so the migration won't work.

Reply:
A `--strip-oci-ta-suffix` flag would have the same problem, they can still forget to use it. Which is why I'd prefer that stripping the suffix be the default behavior. But it's at least a bit easier than having to write out a whole pipeline task name

Reply:
Then, we can have this behaviors:

* if users give an actual task name, e.g. `sast-coverity-check-oci-ta`, and the `--pipeline-task-name` is omitted, the pipeline task name is set to the string by stripping suffix `-oci-ta`.
* specifying `--pipeline-task-name` can overwrite this behavior and use the given name.